synapse/models/risk.py

```diff
@@ -106,19 +106,20 @@ def getModelDefs(self):
                         ),
                     },
                 }),
-                ('risk:attacktype', ('taxonomy', {}), {
-                    'doc': 'A taxonomy of attack types.',
+                ('risk:attack:type:taxonomy', ('taxonomy', {}), {
                     'interfaces': ('meta:taxonomy',),
+                    'doc': 'A hierarchical taxonomy of attack types.',
                 }),
-                ('risk:compromisetype', ('taxonomy', {}), {
-                    'doc': 'A taxonomy of compromise types.',
+                ('risk:compromise:type:taxonomy', ('taxonomy', {}), {
                     'ex': 'cno.breach',
                     'interfaces': ('meta:taxonomy',),
+                    'doc': 'A hierarchical taxonomy of compromise types.',
                 }),
-                ('risk:tool:software:taxonomy', ('taxonomy', {}), {
-                    'doc': 'A taxonomy of software / tool types.',
+                ('risk:tool:software:type:taxonomy', ('taxonomy', {}), {
                     'interfaces': ('meta:taxonomy',),
+                    'doc': 'A hierarchical taxonomy of software tool types.',
                 }),
+                # FIXME
                 ('risk:availability', ('taxonomy', {}), {
```

Comment:
Already tagged #FIXME but can see risk:availability:type:taxonomy for the type and risk:tool:software:availability:type for the prop (to deconflict w/risk:tool:software:type for the tool type generally).

Reply:
We can discuss. Not every use of a taxonomy is for a "type" but this one needed to be discussed in any case :)